### PR TITLE
Create Playwright test harness

### PR DESCRIPTION
**Pull request by HannahZhuSWE**
Created a playwright test harness and github action that runs the tests on push and pull requests to the main branch and at midnight. Tests can be run with `npm run playwright_test`. Note in order to run playwright a package.json file and start script was created.

### CODE REVIEW

**Comment by HannahZhuSWE** [comment]
@nitya

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x]
```
Comment (fhinkel) [inline .github/workflows/playwright.yml:28]: Any reason not to include 18?

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+    - uses: actions/checkout@v2
```
Comment (fhinkel) [inline .github/workflows/playwright.yml:32]: Can we use v3 for checkout and setup-node?

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - uses: actions/setup-node@v2
+      with:
+        node-version: '14'
```
Comment (fhinkel) [inline .github/workflows/playwright.yml:35]: Is this overwriting the versions specified in the matrix?

File: .github/workflows/playwright.yml (hunk 1 above)
Comment (HannahZhuSWE) [reply .github/workflows/playwright.yml:28]: No reason that I'm aware of, I'll add 18 in

File: .github/workflows/playwright.yml (hunk 2 above)
Comment (HannahZhuSWE) [reply .github/workflows/playwright.yml]: yup I'll change that right away

**Review by fhinkel — CHANGES_REQUESTED** [review]
Thank you! 👏 

We should include Node18 and use v3 for checkout and setup-node. 

For the PR title, I'd prefer imperative present tense, e.g., `Create Playwright test harness`. That's inline with the commit history and common best practice, see https://365git.tumblr.com/post/3308646748/writing-git-commit-messages

File: .github/workflows/playwright.yml (hunk 3 above)
Comment (HannahZhuSWE) [reply .github/workflows/playwright.yml]: Oops I missed that, I just changed it in the new commit

**Comment by fhinkel** [comment]
@manekinekko  can you look into why the CI is not running on this PR please?

File: .github/workflows/playwright.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Playwright Tests
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
```
Comment (manekinekko) [inline .github/workflows/playwright.yml:8]: In order to run the CI on any branch or PR, we need to change the config to:

```
on: [push, pull_request]
```

**Comment by manekinekko** [comment]
Thank you @HannahZhuSWE for your contribution. I left a comment to change the workflow setting for the CI to run on every branch and PR.

**Comment by HannahZhuSWE** [comment]
Thank you Wassim!

**Comment by HannahZhuSWE** [comment]
@fhinkel I implemented all the suggestions for this PR

**Review by fhinkel — APPROVED** [review]

**Pull request merged**